### DISCUSSION TITLE
인수인계 문서

### DISCUSSION THREAD

**Discussion by ghost** [team-mokcha/corabal-beta#60] [주간 회의록]
## 들어가기 앞서
- 이 문서는 혜진님께서 29일에 궁금하다고 말씀하신 내용 + 제가 따로 인수인계 드리고 싶은 것들 위주로 작성되었습니다.
- 제가 커버하지 못한 내용은 언제든 말씀해주시면 확인되는 대로 답변 드리겠습니다.

### 1. JSX.Element로 함수 컴포넌트의 리턴 타입을 지정한 이유
- 당시에 React.FC vs. JSX.Element 에 대해 고민했었고 React.FC는 defualtProps 지정이 어렵다고 해서 JSX.Element를 썼었어요.
- 근데 지금 다시 리팩토링을 한다면 어차피 defaultProps를 따로 쓰지 않을 것 같으니 React.FC를 사용할 것 같습니다.

### 2. Main 영역 등에서 특정 라이브러리를 선택한 이유
- Divider 같은 경우엔 react-native-elements 라이브러리가 설치되어 있었기 때문에 추가적인 라이브러리 설치 없이 사용했습니다.
- 라이브러리는 어떤 걸 말씀하시는지 몰라서 일단 다 봤는데 혹시 react-native-progress 를 말씀하시는 건가요?
- 그 땐 progress bar를 직접 만들면 시간이 너무 오래 걸릴 것 같아서 우선 라이브러리 사용하고 추후에 바꾸기로 했던 걸로 기억해요.

### 3. type 대신 interface를 사용한 이유
- circleProgressProps에서 interface를 사용했었는데 type으로 바꾸는 게 좋아보입니다.
- 당시에는 interface가 type보다 좀 더 확장성이 좋다고 생각해서 사용했었어요. 

### 1. ProfileIcon 컴포넌트의 margin 값 고치기
- 고양이 얼굴이 프로필, 헤더, 샵 쪽에서 사용되는데 당시 프로필 스크린을 리팩토링 하면서 프로필에서 쓰이는 위치만 수정해놨어요.
- 주석으로 적은 margin 값 고치기는 그래서 적어놓은 내용입니다.

### 2. 광고
- 리워드형 광고는 [구글 애드몹](https://admob.google.com/intl/ko/home/) 이용했고, `handle-ads.tsx` 쪽에서 광고 보여주는 로직이 있습니다. 
- 현재 들어가있는 아이디 "ca-app-pub-3940256099942544/5224354917"는 테스트용이니 런칭 시엔 커라밸 계정으로 전환하셔야 해요.

### 3. 그 외
- github organization, notion, discord channel 등도 혜진님이 이 문서를 확인하시는 대로 정리하겠습니다.
- 그동안 같이 프로젝트를 진행할 수 있어서 정말 좋았고, 끝까지 참여하진 못했지만 앱 런칭되면 꼭 주변에 홍보 많이 할게요. 새해 복 많이 받으세요!

**Comment by HyejinYang** [comment]
@alveloper 문서 확인했습니다. 수연님도 그간 고생 많으셨어요. 새해 복 많이 받으세요!
